src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -49,6 +63,49 @@ module.exports = function<T, P, I>(config : HostConfig<T, P, I>) : Reconciler {
 
   let nextUnitOfWork : ?Fiber = null;
 
+  let currentRootsWithPendingWork : ?Fiber = null;
+
+  function findNextUnitOfWork(priorityLevel : PriorityLevel) : ?Fiber {
+    let current = currentRootsWithPendingWork;
+    while (current) {
+      if (current.pendingWorkPriority !== 0 &&
+          current.pendingWorkPriority <= priorityLevel) {
+        // This node has work to do that fits our priority level criteria.
+        if (current.pendingProps !== null) {
+          // We found some work to do. We need to return the "work in progress"
+          // of this node which will be the alternate.
+          return current.alternate;
+        }
+        // If we have a child let's see if any of our children has work to do.
+        // Only bother doing this at all if the current priority level matches
+        // because it is the highest priority for the whole subtree.
+        if (current.child) {
+          current = current.child;
+          continue;
+        }
+        // If we match the priority but has no child and no work to do,
+        // then we can safely reset the flag.
+        current.pendingWorkPriority = NoWork;
```

Comment:
So this is where the priority gets cleared usually? I would have expected it to happen in CompleteWork or something like that. Is there an advantage to putting it here?